md-dao/src/main/java/com/wipro/ats/bdre/md/dao/ProcessTemplateDAO.java

```diff
@@ -590,8 +590,8 @@ public void delete(Integer id) {
                     propertiesList.add(properties);
                 }
 
-                if (propTempKeyList.size() != 0) {
-                    Criteria fetchMissingKeyPropertiesList = session.createCriteria(com.wipro.ats.bdre.md.dao.jpa.Properties.class).add(Restrictions.eq("id.processId", processId)).add(Restrictions.not(Restrictions.in("id.propKey", propTempKeyList)));
+                if (!propTempKeyList.isEmpty()) {
+                    Criteria fetchMissingKeyPropertiesList = session.createCriteria(com.wipro.ats.bdre.md.dao.jpa.Properties.class).add(Restrictions.eq(CONFIGGROUPID, processId)).add(Restrictions.not(Restrictions.in("id.propKey", propTempKeyList)));
                     List<com.wipro.ats.bdre.md.dao.jpa.Properties> jpaKeyPropertiesList = fetchMissingKeyPropertiesList.list();
                     for (com.wipro.ats.bdre.md.dao.jpa.Properties jpaProperty : jpaKeyPropertiesList) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)